### PR TITLE
issue-22: leader board

### PR DESCRIPTION
**Pull request by Bo2208**

### CODE REVIEW

File: packages/app-lixi/src/components/Posts/PostsListing.tsx [hunk 1]
```diff
@@ -131,6 +132,11 @@ const PostsListing: React.FC<PostsListingProps> = ({ className }: PostsListingPr
   const [queryPostTrigger, queryPostResult] = useLazyPostQuery();
   const latestBurnForPost = useAppSelector(getLatestBurnForPost);
   const filterValue = useAppSelector(getFilterPostsHome);
+  const getTopFirst = useAppSelector(getLeaderBoard);
+
+  // console.log(getTopFirst);
```
Comment (khanhvinhnguyen) [inline packages/app-lixi/src/components/Posts/PostsListing.tsx:137]: please remove console.log

File: packages/app-lixi/src/containers/Sidebar/SideBarRanking.tsx [hunk 2]
```diff
@@ -18,25 +18,33 @@ import intl from 'react-intl-universal';
...
   href?: string;
   onClickItem?: () => void;
 }) => (
   <Link onClick={onClickItem} href={href}>
     <a>
       <Space className={'item-access'}>
-        <img style={{ borderRadius: '50%' }} width={48} height={48} src={icon ? icon : '/images/xpi.svg'} alt="" />
-        <p style={{ margin: '0', color: '#1E1A1D', fontSize: '16px', letterSpacing: '0.5px' }}>{text}</p>
+        <AvatarUser name={text} isMarginRight={false} />
+        <div style={{ margin: '0', color: '#1E1A1D', fontSize: '16px', letterSpacing: '0.5px' }}>
```
Comment (khanhvinhnguyen) [inline packages/app-lixi/src/containers/Sidebar/SideBarRanking.tsx:42]: Consider to use styled-component so it's more flexible and can add responsive easier

File: packages/app-lixi-api/src/modules/core/account/account.controller.ts [hunk 3]
```diff
@@ -113,6 +115,64 @@ export class AccountController {
...
+        return { ...obj1, ...obj2 };
+      });
+
+      const result = accountDDO.map(data => _.omit({ ...data }, 'publicAddress'))
+      return result
+    } catch (err: unknown) {
+      if (err instanceof VError) {
+        throw new HttpException(err, HttpStatus.INTERNAL_SERVER_ERROR);
+      } else {
+        const unableGetAccountMessage = await i18n.t('account.messages.unableGetAccount');
```
Comment (khanhvinhnguyen) [inline packages/app-lixi-api/src/modules/core/account/account.controller.ts:170]: why unableGetAccount while we getMostBurnAccount?
Is the message correct?

File: packages/app-lixi/src/store/account/api.ts [hunk 4]
```diff
@@ -131,7 +131,20 @@ const accountApi = {
         const { response } = err;
         throw response?.data ?? err ?? 'Network Error';
       });
-  }
+  },
+
+  getTopFive() {
+    const url = '/api/accounts/top5';
```
Comment (vince8x) [inline packages/app-lixi/src/store/account/api.ts:137]: We should use the url 'api/accounts/leaderboard?first=5

File: packages/app-lixi/src/containers/Sidebar/SideBarRanking.tsx [hunk 5]
```diff
@@ -464,7 +464,7 @@ const SidebarRanking = () => {
               <Tabs.TabPane tab={<ShopOutlined />} key="page">
                 <div className="content">
                   <h3>Top Pages</h3>
-                  {randomShortCut &&
+                  {/* {randomShortCut &&
```
Comment (vince8x) [inline packages/app-lixi/src/containers/Sidebar/SideBarRanking.tsx:467]: We can remove the code if not use

File: packages/app-lixi-api/src/modules/core/account/account.controller.ts [hunk 6]
```diff
@@ -113,6 +115,64 @@ export class AccountController {
...
+        const publicAddress = this.XPI.Address.toXAddress(legacyAddress);
+
+        const totalBurned = account._sum.burnedValue
+        return {
+          publicAddress,
+          totalBurned
+        }
+      })
+
+      const accountAddress = _.map(accounts, 'publicAddress');
```
Comment (vince8x) [inline packages/app-lixi-api/src/modules/core/account/account.controller.ts:149]: Should be named accountAddresses

File: packages/app-lixi-api/src/modules/core/account/account.controller.ts [hunk 7]
```diff
@@ -113,6 +115,64 @@ export class AccountController {
...
+        const legacyAddress = this.XPI.Address.hash160ToLegacy(burnedBy)
+
+        const publicAddress = this.XPI.Address.toXAddress(legacyAddress);
+
+        const totalBurned = account._sum.burnedValue
+        return {
+          publicAddress,
+          totalBurned
+        }
+      })
```
Comment (vince8x) [inline packages/app-lixi-api/src/modules/core/account/account.controller.ts:147]: Missing semicolon here?

File: packages/app-lixi-api/src/modules/core/account/account.controller.ts [hunk 8]
```diff
@@ -113,6 +115,64 @@ export class AccountController {
     }
   }
 
+  @Get('top5')
+  async getMostBurnAcount() {
+    try {
+      const accountsBuffer = await this.prisma.burn.groupBy({
```
Comment (vince8x) [inline packages/app-lixi-api/src/modules/core/account/account.controller.ts:121]: We can just name it "leaderboardAccounts" here

File: packages/app-lixi-api/src/modules/core/account/account.controller.ts [hunk 9]
```diff
@@ -113,6 +115,64 @@ export class AccountController {
...
+        },
+        orderBy: {
+          _sum: {
+            burnedValue: 'desc'
+          }
+        },
+        take: 5
+      });
+
+      const accounts = accountsBuffer.map((account: any) => {
```
Comment (vince8x) [inline packages/app-lixi-api/src/modules/core/account/account.controller.ts:134]: We could name "addressAndTotalBurntArray"

File: .vscode/launch.json [hunk 10]
```diff
@@ -0,0 +1,20 @@
+{
```
Comment (vince8x) [inline .vscode/launch.json:1]: Please ignore the custom .vscode debug profile

File: packages/app-lixi-api/src/modules/core/account/account.controller.ts [hunk 11]
```diff
@@ -113,6 +117,65 @@ export class AccountController {
     }
   }
 
+  @Get('leaderboard')
+  async getMostBurnedAcount(@Query('topMostBurned') topMostBurned: number, @I18n() i18n: I18nContext) {
```
Comment (vince8x) [inline packages/app-lixi-api/src/modules/core/account/account.controller.ts:121]: The method name should be getLeaderboard.
The query param name should be limit or take.

File: packages/app-lixi/src/store/account/api.ts [hunk 12]
```diff
@@ -131,7 +131,24 @@ const accountApi = {
...
       });
-  }
+  },
+
+  getLeaderboard() {
+    const url = 'api/accounts/leaderboard';
+    return axiosClient
+      .get(url, {
+        params: {
+          topMostBurned: 5
```
Comment (vince8x) [inline packages/app-lixi/src/store/account/api.ts:141]: Param should be name "limit"

File: packages/app-lixi/src/store/account/saga.ts [hunk 13]
```diff
@@ -107,6 +110,34 @@ function* generateAccountSaga(action: PayloadAction) {
   yield put(postAccount(account));
 }
 
+function* getLeaderboardSaga(action: PayloadAction<number>) {
+  try {
+    yield put(showLoading(getLeaderboard.type));
+    const data = yield call(accountApi.getLeaderboard);
+    // yield put(getAccountSuccess(data));
```
Comment (vince8x) [inline packages/app-lixi/src/store/account/saga.ts:117]: Please remove unused code

File: packages/app-lixi/src/store/account/state.ts [hunk 14]
```diff
@@ -12,5 +12,6 @@ export interface AccountsState extends EntityState<Account> {
   pageAvatarUpload: Upload;
   postCoverUploads: Upload[];
   editorCache: Nullable<string> | undefined;
+  leaderBoard: Array<any>;
```
Comment (vince8x) [inline packages/app-lixi/src/store/account/state.ts:15]: We should not use any. We can use Array<Account>

File: packages/app-lixi/src/containers/Sidebar/SideBarRanking.tsx [hunk 15]
```diff
@@ -303,6 +327,10 @@ const SidebarRanking = () => {
   });
   const { validateMnemonic } = Wallet;
 
+  useEffect(() => dispatch(getLeaderboard()), []);
+
+  console.log(getTops)
```
Comment (vince8x) [inline packages/app-lixi/src/containers/Sidebar/SideBarRanking.tsx:332]: Please remove unsued debug code

File: packages/app-lixi-api/src/modules/core/account/account.controller.ts [hunk 16]
```diff
@@ -113,6 +117,65 @@ export class AccountController {
...
+
+      const accountDTO = await this.prisma.account.findMany({
+        where: {
+          address: {
+            in: accountAddresses
+          }
+        }
+      });
+
+      const accountDDO = addressAndTotalBurntArray.map((account) => {
```
Comment (vince8x) [inline packages/app-lixi-api/src/modules/core/account/account.controller.ts:161]: We could use ```accountDTO.find(account => _.includes(accountAddresses, account.address))```

File: packages/app-lixi/src/components/Posts/PostsListing.tsx [hunk 17]
```diff
@@ -171,6 +172,10 @@ const PostsListing: React.FC<PostsListingProps> = ({ className }: PostsListingPr
   const walletStatus = useAppSelector(getWalletStatus);
   const failQueue = useAppSelector(getFailQueue);
   const filterValue = useAppSelector(getFilterPostsHome);
+  const getTopFirst = useAppSelector(getLeaderBoard);
```
Comment (vince8x) [inline packages/app-lixi/src/components/Posts/PostsListing.tsx:175]: The variable name should be leaderboard

File: packages/app-lixi-api/src/modules/core/account/account.controller.ts [hunk 18]
```diff
@@ -113,6 +117,65 @@ export class AccountController {
     }
   }
 
+  @Get('getleaderboard')
+  async getMostBurnedAcount(@Query('limit') limit: number, @I18n() i18n: I18nContext): Promise<any> {
```
Comment (vince8x) [inline packages/app-lixi-api/src/modules/core/account/account.controller.ts:121]: The method should be getLeaderboard

File: packages/app-lixi-api/src/modules/core/account/account.controller.ts [hunk 19]
```diff
@@ -113,6 +117,65 @@ export class AccountController {
...
+      });
+
+      const result = accountDDO.map(data => _.omit({ ...data }, 'publicAddress'))
+      return result
+    } catch (err: unknown) {
+      console.log(err)
+      if (err instanceof VError) {
+        throw new HttpException(err, HttpStatus.INTERNAL_SERVER_ERROR);
+      } else {
+        const getMostBurnAccount = await i18n.t('account.messages.unableGetAccount');
```
Comment (vince8x) [inline packages/app-lixi-api/src/modules/core/account/account.controller.ts:173]: Here we means that the error message should be could not get leaderboard

**Review by vince8x — APPROVED** [review]

**Pull request merged**